docs/src/modules/components/MarkdownDocs.js

```diff
@@ -0,0 +1,296 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import path from 'path';
+import { useTheme } from '@mui/system';
+import { exactProp } from '@mui/utils';
+import { styled } from '@mui/material/styles';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import Demo from 'docs/src/modules/components/Demo';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import MarkdownElement from 'docs/src/modules/components/MarkdownElement';
+import AppLayoutDocs from 'docs/src/modules/components/AppLayoutDocs';
+import { useTranslate, useUserLanguage } from 'docs/src/modules/utils/i18n';
+import Ad from 'docs/src/modules/components/Ad';
+import AdGuest from 'docs/src/modules/components/AdGuest';
+
+function noComponent(moduleID) {
+  return function NoComponent() {
+    throw new Error(`No demo component provided for '${moduleID}'`);
+  };
+}
+
+const classes = {
+  imageContainer: 'MuiToolpadDocs-ImageContainer',
+  enlargeIcon: 'MuiToolpadDocs-EnlargeIcon',
+  fullSizeImage: 'MuiToolpadDocs-FullSizeImage',
+  fullSizeImageDialog: 'MuiToolpadDocs-FullSizeImageDialog',
+  fullSizeImageDialogTitle: 'MuiToolpadDocs-FullSizeImageDialogTitle',
+  fullSizeImageDialogCloseButton: 'MuiToolpadDocs-FullSizeImageDialogCloseButton',
+};
+
+const Root = styled('div')(
+  ({ theme }) => {
+    const isMdUp = useMediaQuery(theme.breakpoints.up('md'));
+    return {
+      '& .markdown-body': {
+        lineHeight: 1.7,
+      },
+      '& .arcade': {
+        position: 'relative',
+        paddingBottom: 'calc(54.79166666666667% + 41px)',
+        height: 0,
+      },
+      [`& .${classes.imageContainer}`]: {
+        position: 'relative',
+        display: 'inline-block',
+        cursor: 'pointer',
+      },
+
+      '& img, & video': {
+        border: `1px solid ${theme.palette.divider}`,
+        marginTop: theme.spacing(3),
+        marginBottom: theme.spacing(2),
+        marginLeft: 'auto',
+        marginRight: 'auto',
+
+        // Override styles defined in MarkdownElement
+        borderRadius: '4px!important',
+        display: 'block!important',
+      },
+
+      [`& .${classes.imageContainer}:hover .${classes.enlargeIcon}`]: {
+        // breakpoints
+        opacity: isMdUp ? 0.7 : 0,
+      },
+
+      [`& .${classes.enlargeIcon}`]: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%, -50%)',
+        width: 40,
+        height: 40,
+        backgroundImage: 'url(/static/icons/enlarge.svg)',
```

Comment:
Personally, I have find the cursor a better UX

<img width="478" alt="Screenshot 2023-05-20 at 00 10 56" src="https://github.com/mui/mui-toolpad/assets/3165635/aebea076-d2b5-4283-82e8-42f2915d8a31">

https://medium.com/@justincloud/i-didnt-like-the-price-so-i-built-a-competitor-over-a-weekend-here-is-what-happened-6309a2a3a761

https://github.com/francoischalifour/medium-zoom